Mathlib/LinearAlgebra/TensorProduct/Basis.lean

```diff
@@ -66,5 +67,181 @@ lemma Basis.baseChange_apply (b : Basis ι R M) (i) :
     b.baseChange S i = 1 ⊗ₜ b i := by
   simp [Basis.baseChange, Basis.tensorProduct]
 
+variable (K : Type*) [CommSemiring K] (B C: Type*) [AddCommMonoid B] [Module K B]
+    [AddCommMonoid C] [Module K C]
+
+/-- Elements in B ⊗ C can be represented by sum of elements in B tensor elements of basis of C.-/
+lemma TensorProduct.eq_repr_basis_right
+    {ιC ιB : Type*} (𝒞 : Basis ιC K C) (ℬ : Basis ιB K B) (x : B ⊗[K] C) :
+    ∃ (s : Finset ιC) (b : ιC → B), ∑ i ∈ s, b i ⊗ₜ[K] 𝒞 i = x := by
+  classical
+  let 𝒯 := Basis.tensorProduct ℬ 𝒞
+  have eq1 := calc x
+      _ = ∑ ij ∈ (𝒯.repr x).support, (𝒯.repr x) ij • 𝒯 ij := 𝒯.linearCombination_repr x |>.symm
+      _ = ∑ ij ∈ (𝒯.repr x).support, (𝒯.repr x) (ij.1, ij.2) • 𝒯 (ij.1, ij.2) :=
+          Finset.sum_congr rfl <| by simp
+      _ = ∑ i ∈ (𝒯.repr x).support.image Prod.fst, ∑ j ∈ (𝒯.repr x).support.image Prod.snd,
+            𝒯.repr x (i, j) • 𝒯 (i, j) := by
+          rw [← Finset.sum_product']
+          apply Finset.sum_subset
+          · rintro ⟨i, j⟩ hij
+            simp only [Finsupp.mem_support_iff, ne_eq, Finset.mem_product, Finset.mem_image,
+              Prod.exists, exists_and_right, exists_eq_right, Subtype.exists, 𝒯] at hij ⊢
+            exact ⟨⟨j, hij⟩, ⟨i, hij⟩⟩
+          · rintro ⟨i, j⟩ hij1 hij2
+            simp only [Finset.mem_product, Finset.mem_image, Finsupp.mem_support_iff, ne_eq,
+              Prod.exists, exists_and_right, exists_eq_right, Subtype.exists, Decidable.not_not,
+              Basis.tensorProduct_apply, smul_eq_zero, 𝒯] at hij1 hij2 ⊢
+            rw [hij2]
+            simp only [zero_smul]
+      _ = ∑ j ∈ (𝒯.repr x).support.image Prod.snd, ∑ i ∈ (𝒯.repr x).support.image Prod.fst,
+            𝒯.repr x (i, j) • 𝒯 (i, j) := Finset.sum_comm
+      _ = ∑ j ∈ (𝒯.repr x).support.image Prod.snd, ∑ i ∈ (𝒯.repr x).support.image Prod.fst,
+            𝒯.repr x (i, j) • (ℬ i ⊗ₜ[K] 𝒞 j) := by
+          refine Finset.sum_congr rfl fun _ _ => ?_
+          simp only [𝒯, Basis.tensorProduct_apply]
+      _ =  ∑ j ∈ (𝒯.repr x).support.image Prod.snd, ∑ i ∈ (𝒯.repr x).support.image Prod.fst,
+            (𝒯.repr x (i, j) • ℬ i) ⊗ₜ[K] 𝒞 j := by
+          refine Finset.sum_congr rfl fun _ _ => Finset.sum_congr rfl fun _ _ => ?_
+          rw [TensorProduct.smul_tmul']
+      _ =  ∑ j ∈ (𝒯.repr x).support.image Prod.snd, (∑ i ∈ (𝒯.repr x).support.image Prod.fst,
+            (𝒯.repr x (i, j) • ℬ i)) ⊗ₜ[K] 𝒞 j := by
+          refine Finset.sum_congr rfl fun _ _ => ?_
+          rw [TensorProduct.sum_tmul]
+  exact ⟨_, _, eq1.symm⟩
+
+/-- Elements in B ⊗ C can be represented by sum of elements of basis of B tensor elements of C.-/
+lemma TensorProduct.eq_repr_basis_left
+    {ιB ιC: Type*} (ℬ : Basis ιB K B) (𝒞 : Basis ιC K C) (x : B ⊗[K] C) :
+    ∃ (s : Finset ιB) (c : ιB → C), ∑ i ∈ s, ℬ i ⊗ₜ[K] c i = x := by
```

Comment:
K is commutative so surely you can just deduce this from the previous lemma via [`Algebra.TensorProduct.comm`](https://leanprover-community.github.io/mathlib4_docs/Mathlib/RingTheory/TensorProduct/Basic.html#Algebra.TensorProduct.comm) rather than going through the long proof again.